planet/subscription_request.py

```diff
@@ -77,31 +92,36 @@ def build_request(name: str,
 
     ```
 
-    Parameters:
-        name: Name of the subscription.
-        source: A source for the subscription, i.e. catalog.
-        delivery: A delivery mechanism e.g. GCS, AWS, Azure, or OCS.
-        notifications: Specify notifications via email/webhook.
-        tools: Tools to apply to the products. Order defines
-            the toolchain order of operatations.
     """
-    details = {"name": name, "source": source, "delivery": delivery}
+    details = {
+        "name": name, "source": dict(source), "delivery": dict(delivery)
```

Comment:
Here is the consequence of accepting mappings: we need to explicitly convert them to dicts here. The upside is that we will never modify dicts that users pass in to this function, which can cause all kinds of subtle bugs.

Reply:
We copy information now, which has some overhead, but makes calling code more reliable. And building a subscriptions request doesn't need to be super optimized.